### PR TITLE
Fix install rtools

### LINKED ISSUE #148: Error with install.Rtools()
**Describe the bug**
`installr::install.Rtools()` results in an error: 'Couldn't find a table'

**To Reproduce**
`installr::install.Rtools()`

I get this:
```
No encoding supplied: defaulting to UTF-8.
Argument 'which' was left unspecified. Choosing first table.
Error: Couldn't find a table.
```

 ```
sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.0.0 tools_4.0.0 

```

### PR DESCRIPTION
**Pull request by burgerga**
Hi @talgalili, 

Fixes #148 

I think with these changes `install.Rtools` will be a lot simpler and a little more robust :)
Unfortunately, it's not backward compatible with the old `install.Rtools` so if you're using semantic versioning you need to bump the major version.

Rational behind some (possibly controversial) changes:
* Removal of `choose` option: Much simpler code (no more parsing of this htmltab), and I think most users simply want to install a compatible Rtools and not some historic Rtools that won't work with their setup anyway. If not, they're most likely advanced users that can find the appropriate installer on their own 😉  (for example with `install.URL(installr:::getRtoolsUrl("3.0"))`
* Add `pkgbuild` to `Depends`. Since this code depends on pkbuild > 1.0.8 that can now be enforced (I don't know what the next version of `pkgbuild` will be or when it hits CRAN)
* Added simpler functions to check for latest R version/update, you might want to use them in the `check.for.updates.R` function but at the moment you also use a date there which is not reported by this simple functions.

**NB:** Won't work until pkgbuild > 1.0.8 will hit CRAN, to get it to work use `remotes::install_github('r-lib/pkgbuild')`

Looking forward to your thoughts!
Gerhard

### CODE REVIEW

**Comment by talgalili** [comment]
Hey Gerhard,
I think the first priority of installr is to allow people to upgrade their
R installation, so placing such a strong dependency sounds excessive.
What we can do is to make the packages you mention be in "suggested", and
have inside the function install.rtools a condition in the beginning that
requires the dependencies you mentioned, which if the user doesn't have
then there will be a message with instructions on what to run/install (or
the option to do it automatically for the user).
This will also keep us within the 0.x.y version branch (I'd like to move to
version 1.x.y only if installr would start including upgrade code for Mac
OS or Linux).

WDYT?

**Comment by burgerga** [comment]
Woops! My bad, it was supposed to go in `Imports`, `Depends` is indeed a bit excessive ;)
Does that change it for you?  

Also I didn't realize `installr` was still in 0.x.y, according to https://semver.org/#spec-item-4 we can safely stay in 0.x.y even if we change the API, so that's fine as well. (BTW even if you keep `pkgload` in `Suggests` this PR changes the public API, because the function signature of `install.Rtools` has changed)

**PS** I think it would be fine to release a 1.x.y; Linux and Mac support are very big and involved features that you can each give their separate major release. (And as a long time Linux user, I would **never ever** trust an R package to install Linux software for me 😀)

**Comment by burgerga** [comment]
By the way, that Depends -> Imports holds for `stringr` and `utils` as well, see http://r-pkgs.had.co.nz/namespace.html#search-path.

**Comment by talgalili** [comment]
Fair points, I guess that moving to 1.x.y is more or less in order.
Regarding depends vs imports, they are the same in terms of constraints
(the only difference is if the package must be loaded to the user namespace
when called. In that sense, people often prefer imports vs depends since it
helps keep the user global environment cleaner).

As for utils and stringr, I agree - these were a poor design decisions
earlier on that I just didn't get around to fix. I guess doing so would be
a good move before moving to 1.x.y :)

In any event, if you're happy to go with suggests, I'd be happy for such a
PR.

T

**Comment by burgerga** [comment]
I'm personally not very fond of `Suggests` in this case. Stopping function execution to instruct the user to install another package seems very much against the spirit of `installr` 😉 . I think this a perfect fit for Imports; it's a small (136KB) package developed by the RStudio people. Besides, I don't think there is anything wrong with having a few Imports (c.f [ggplot2](https://github.com/tidyverse/ggplot2/blob/master/DESCRIPTION)), and we're still very far away from [devtools levels of Imports](https://github.com/r-lib/devtools/blob/master/DESCRIPTION) 😛 . 
But if you really want to use `Suggests`, let me know, maybe we can change `require2` to take a version constraint into account (something like `require2('pkgbuild', version = "> 1.0.8")`.

**Comment by talgalili** [comment]
I think that since the main usecase of installr is update R (not installing
rtools), I think we should keep pkgbuild as a suggest.
Looking at it:
https://cran.r-project.org/web/packages/pkgbuild/index.html
I see it has a bunch of imports itself, including major packages such as R6
And it requires R 3.1, while installr currently should work with R 2.14
(that's of course not true, since I need to remove the dependency of
installr on stringr, but that is a doable task).

WDYT?

**Comment by burgerga** [comment]
Sure, give me time to make some changes.

Have you had complaints that installr doesn't work with R < 3.1 (April 2014)? I think most users of installr are updating from fairly recent versions (e.g. 3.6 -> 4.0 or 4.0.0 -> 4.0.2). Of course, it's possible to remove stringr, but I don't see the benefits, it's just extra work for you and potential contributors.

**Comment by talgalili** [comment]
My concern is that users of older versions of R wouldn't get around to
bother and report this issue, since they wouldn't even get to run the CRAN
install.
I don't have a metric on how many R users are using each version, do you
know of such a data source?

**Comment by burgerga** [comment]
Yes, I suspect that could be the case. I'm also very curious about R usage per version. 
I can't imagine using R without RStudio and ggplot (maybe I'm spoiled :P), and they have R >= 3.0.1, and R >= 3.2

**Comment by talgalili** [comment]
After thinking about this a bit more, I see that R 3.1 was released in 2014. So my guess is that if someone didn't update their R version in the past 6 years or so, they might as well do it manually just to be informed that r-project has a new website :)
o.k., I'll take-in your pull request. Do you have more changes that you're planning to introduce?
(for example, using under_scores for variable names insteadOfusing other.formats would be nice :) )

**Comment by talgalili** [comment]
(and yes, I know the current installr has many odd combinations of dots and upper/lower cases in function names. I'm trying to move towards a nice future here :) )

**Comment by burgerga** [comment]
Good to hear! It was still on my todo list, but I was swamped preparing a course I'm teaching after the summer 😞 
I was planning to move `pkgbuild` back to `Suggests`, and to modify the `require2` function to take a `min_version` argument (or something).
And it's never too late to introduce some style 🤵 (I see I made quite the mess indeed)
Do you want https://style.tidyverse.org/? (I was quite happy with `getLatestRversion`'s similarity to `base::getRVersion`, but changing it is not a big deal 😉)

**Comment by talgalili** [comment]
Thanks, yes
https://style.tidyverse.org/
Is great :)

**Comment by talgalili** [comment]
Hi @burgerga - any update?

**Comment by burgerga** [comment]
No, sorry, still swamped in work. Will try to have a look this week

**Comment by talgalili** [comment]
Sounds good, thanks :)

>

**Comment by burgerga** [comment]
Finally had some time to finish this up, also closes #158

File: R/installRtools.R [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+  }
+
+  # Use pkgbuild to check if matching rtools is installed
+  if (check) {
+    require2("pkgbuild")
+    found_rtools <- pkgbuild::has_rtools()
+    if (found_rtools) {
+      cat("No need to install Rtools - You've got the relevant version of Rtools installed\n")
+      return(invisible(FALSE))
```
Comment (burgerga) [inline R/installRtools.R:72]: Shouldn't this be TRUE (a compatible Rtools is available)

File: R/installRtools.R [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+#' install.Rtools(checkRupdate = F)
+#' }
+install.Rtools <- function(check = TRUE, check_r_update = TRUE, GUI = TRUE, ...) {
+  if (check_r_update && r_update_available(ignore_patchlevel = T)) {
+    update_r <- ask.user.yn.question(
+      "A newer R version is available, do you want to update R first?", GUI = GUI)
+    if (update_r) {
+      updateR(GUI = GUI, print_R_versions = F, install_R = T, start_new_R = T, quit_R = F)
+      print("Please run installr::install.Rtools using the newly installed R")
+      return(FALSE)
```
Comment (burgerga) [inline R/installRtools.R:62]: I see I forgot an invisible here...

**Comment by burgerga** [comment]
I also used `styler` and `lintr` to apply tidyverse style to the code I created/changed. That's also where the `.lintr` file comes from. Currently, I have set a max length of 100 characters, but maybe you want a different max line length (80 is really outdated though :P)

**Comment by burgerga** [comment]
Also maybe we should offer to put RStudio on the path as suggested here: https://cran.r-project.org/bin/windows/Rtools/? (or at least point to that page for post-installation steps)

File: R/installRtools.R (hunk 2 above)
Comment (talgalili) [reply R/installRtools.R:62]: You can also keep it without invisible

File: R/installRtools.R (hunk 1 above)
Comment (talgalili) [reply R/installRtools.R:72]: Interesting question of definitions. Should the return be if the software is available, or if it was installed. I think it should be "was it installed". Hence, this should probably be FALSE. But I'm willing to rethink this.

**Comment by talgalili** [comment]
The diff looks good to me, so I'm merging. Thanks.

**Comment by talgalili** [comment]
As for the other ideas - feel free to add them as a new diff.

**Pull request merged**